cartridges/custom_cartridge/cartridge/controllers/CheckoutServices.js

```diff
@@ -0,0 +1,594 @@
+            // respond with form data and errors
+            res.json({
+                form: paymentForm,
+                fieldErrors: formFieldErrors,
+                serverErrors: paymentFormResult.serverErrors ? paymentFormResult.serverErrors : [],
+                error: true
+            });
+            return next();
+        }
+
+        res.setViewData(paymentFormResult.viewData);
+
+        this.on('route:BeforeComplete', function (req, res) { // eslint-disable-line no-shadow
+            var BasketMgr = require('dw/order/BasketMgr');
+            var HookMgr = require('dw/system/HookMgr');
+            var PaymentMgr = require('dw/order/PaymentMgr');
+            var Transaction = require('dw/system/Transaction');
+            var AccountModel = require('*/cartridge/models/account');
+            var OrderModel = require('*/cartridge/models/order');
+            var URLUtils = require('dw/web/URLUtils');
+            var Locale = require('dw/util/Locale');
+            var basketCalculationHelpers = require('*/cartridge/scripts/helpers/basketCalculationHelpers');
+            var hooksHelper = require('*/cartridge/scripts/helpers/hooks');
+            var validationHelpers = require('*/cartridge/scripts/helpers/basketValidationHelpers');
+
+            var currentBasket = BasketMgr.getCurrentBasket();
+
+            var billingData = res.getViewData();
+
+            if (!currentBasket) {
+                delete billingData.paymentInformation;
+
+                res.json({
+                    error: true,
+                    cartError: true,
+                    fieldErrors: [],
+                    serverErrors: [],
+                    redirectUrl: URLUtils.url('Cart-Show').toString()
+                });
+                return;
+            }
+
+            //add shipping addresses and billing address to external service
+            var customer = req.currentCustomer.raw.profile;
+            var billingAddress = currentBasket.billingAddress;
+            var result;
+            if (customer) {
+                var customerIntegrateId = customer.custom.v_integrateId;
+                var HookManager = require('dw/system/HookMgr');
+                if (HookManager.hasHook('app.register.requestCustomerToExternalService')) {
+                    result = HookManager.callHook(
+                        'app.register.requestCustomerToExternalService',
+                        'addCustomersDataFromCheckoutToExternalService',
+                        customerIntegrateId,
+                        currentBasket.shipments,
+                        billingAddress
+                    );
+                }
+                if (!result.ok) {
+                    return;
+                }
+            } else {
+                var customerEmail = currentBasket.customerEmail
+                var HookManager = require('dw/system/HookMgr');
+                if (HookManager.hasHook('app.register.requestCustomerToExternalService')) {
```

Comment:
if and else hooks seems quite similar and can be combined. Otherwise you will have repeated code which should be avoided.

Reply:
That hook is moved in checkoutHelpers and if and else is logic is changed